libraries/botbuilder-dialogs-adaptive/src/adaptiveDialogBot.ts

```diff
@@ -103,6 +102,19 @@ export class AdaptiveDialogBot extends ActivityHandler {
 
         context.turnState.set(languagePolicyKey, this.languagePolicy);
         context.turnState.set(BotTelemetryClientKey, this.telemetryClient);
+
+        // Catch "setTestOptions" event and save into "conversation.testOptions".
+        // Note: This is consumed by AdaptiveExpressions Extensions.RandomNext
+        if (context.activity.type === ActivityTypes.Event && context.activity.name === 'setTestOptions') {
+            console.log('setTestOptions received. This could change the behavior of AdaptiveExpressions RandomNext');
+
+            await context.turnState
```

Comment:
Are any of these calls actually async? I thought all our turn state manipulations were sync.

Reply:
[Used to be](https://github.com/microsoft/botbuilder-js/pull/3792/files#diff-e3423f1173c51c9f256335a75a2705c495a2b163cbbb1291925708cbda444d26R5)

Reply:
Only `set()`
![image](https://user-images.githubusercontent.com/40401643/123170742-baf37180-d42f-11eb-9ebc-3729f2c38743.png)
[Same w/ .NET](https://github.com/microsoft/botbuilder-dotnet/pull/5711/files#diff-f9969fa5df32f8348ae2e61b9a0a0e9f4c4eba9c4ef722dc167e552a30c80748R140)

Reply:
fair! The `setUpTurnState` method is async in .NET so even better to be closer to parity :)

Reply:
It was this `setTestOptions` feature/add that made it `async`, too.